src/cdomain/value/cdomains/int/intervalDomain.ml

```diff
@@ -292,16 +294,13 @@ struct
     match x, y with
     | None, None -> (bot (),{underflow=false; overflow=false})
     | None, _ | _, None -> raise (ArithmeticOnIntegerBot (Printf.sprintf "%s op %s" (show x) (show y)))
-    | (Some (x1,x2) as x), (Some (y1,y2) as y) ->
-      begin
-        let is_zero v = Ints_t.compare v Ints_t.zero = 0 in
-        match y1, y2 with
-        | l, u when is_zero l && is_zero u -> (top_of ik,{underflow=false; overflow=false})
-        | l, _ when is_zero l              -> div ik (Some (x1,x2)) (Some (Ints_t.one,y2))
-        | _, u when is_zero u              -> div ik (Some (x1,x2)) (Some (y1, Ints_t.(neg one)))
-        | _ when leq (of_int ik (Ints_t.zero) |> fst) (Some (y1,y2)) -> (top_of ik,{underflow=false; overflow=false})
```

Comment:
By the way, I'm not sure why, but the old handling of division by 0 is quite inconsistent:
* `x / [0, 0]` returns top (for division by 0).
* `x / [0, 42]` delegates to `x / [1, 42]`, which doesn't return top.
* `x / [-42, 0]` delegates to `x / [-42, -1]`, which doesn't return top.
* `x / [-42, 42]` again returns top (for division by 0).

The new handling consistently returns top in all of the above cases.
Maybe the goal was to imitate the kind of split at 0 which is now part of the general logic in `IArith.div`?